File: Java/BestTimeToBuyAndSellStock.java
package com.example;

import java.util.Scanner;

public class BestTimeToBuyAndSellStock {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		int n = sc.nextInt();
		
		int[] prices = new int[n];
		
		for(int i = 0; i < n; i ++)
		{
			prices[i] = sc.nextInt();
		}
		
		System.out.println(maxProfit(prices));

	}
	
	public static int maxProfit(int[] prices)
	{
		int minPrice = Integer.MAX_VALUE;
		
		int maxProfit = 0;
		
		for(int price : prices)
		{
			if(price < minPrice)
			{
				minPrice = price;
			}
			
			else
			{
				int profit = price - minPrice;
				
				maxProfit = Math.max(maxProfit, profit);
			}
		}
		
		return maxProfit;
	}

}

File: Python/BestTimeToBuyAndSellStock.py
def max_profit(price):

    min_price = float('inf')

    max_profit = 0

    for price in prices:

        if price < min_price:

            min_price = price

        else:

            profit = price - min_price

            max_profit = max(max_profit, profit)

    return max_profit

prices = list(map(int, input().split()))

print(max_profit(prices))